File: src/types/Location.ts
export interface Location {
  id: number
  name: string
  code: string
  type: string
  building: string | null
  floor: string | null
  description: string
  is_active: boolean
  asset_count?: number
  created_at: string
  updated_at: string
}

export interface CreateLocationPayload {
  name: string
  code: string
  type: string
  building?: string
  floor?: string
  description?: string
}

export const LOCATION_TYPES = [
  'terminal', 'gate', 'runway', 'apron', 'hangar',
  'cargo', 'maintenance_bay', 'parking', 'fuel_depot',
  'control_tower', 'office',
] as const

File: backend/schemas/location.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel


class LocationCreate(BaseModel):
    name: str
    code: str
    type: str
    building:    Optional[str] = None
    floor:       Optional[str] = None
    description: Optional[str] = ''


class LocationOut(LocationCreate):
    id:         int
    is_active:  bool
    created_at: datetime
    updated_at: datetime
    model_config = {"from_attributes": True}